mne_bids/write.py

```diff
@@ -2158,3 +2163,400 @@ def write_meg_crosstalk(fname, bids_path, verbose=None):
     logger.info(f'Writing crosstalk file to {out_path}')
     out_path.mkdir()
     shutil.copyfile(src=fname, dst=str(out_path))
+
+
+def _get_daysback(
+    *,
+    bids_path: BIDSPath,
+    rng: np.random.Generator
+) -> int:
+    """Try to find a suitable "daysback" for anonymization."""
+    raw = read_raw_bids(bids_path=bids_path, verbose='error')
+    daysback_min, daysback_max = get_anonymization_daysback([raw])
```

Comment:
using just a single file defeats the purpose of using this function, no? what's min/max ?

Reply:
I do this for performance reasons, e.g. for large datasets on an NAS I don't want to have to perform `read_info()` on hundreds or thousands of files just to determine `daysback`. Therefore, I'm picking one recording and chop off 365 days from the reported `min` and `max`, hoping that this will cover all recordings in the dataset.

Not the greatest solution, and ideas for alternative approaches are welcome!

Reply:
I see, probably won't work for longitudinal datasets but BIDS anonymization is anyway broken for that so I can live with this :)

Reply:
cc @agramfort

Reply:
After discussion with @agramfort I'll now try to come up with an optimized way to take a larger amount of data into account when estimating a good `daysback`. Will push something shortly.

Reply:
My new approach is to pick one run per session for each participant, largely reducing the number of files we need to read in multi-run experiments.

When reading the infos from these files, we display a progress bar if 20 or more files are to be processed. For fewer files, no progress bar is displayed.
(we may wish to adjust this threshold … my idea here was to avoid displaying a progress bar that immediately reaches 100%, cluttering the terminal) Note that this only applies to reading the infos to generate a valid `daysback`. For the actual anonymization, we always display progress bars.